File: algorithms/leet.0299.src.1.js
/**
 * @param {string} secret
 * @param {string} guess
 * @return {string}
 */
var getHint = function(secret, guess) {
    var i, j, x, y, bull, cow, n = secret.length;
    for(i = bull = 0; i < n; ++i) {
        bull += secret[i] == guess[i] ? 1 : 0;
    }
    for(i = 0, cow = -bull; i < 10; ++i) {
        for(j = x = y = 0; j < n; ++j) {
            x += i == secret[j] ? 1 : 0;
            y += i == guess[j] ? 1 : 0;
        }
        cow += Math.min(x, y);
    }
    return ''+bull+'A'+cow+'B';
};

File: algorithms/leet.0299.src.1.py
class Solution:
    def getHint(self, secret: str, guess: str) -> str:
        a = sum([1 if x == y else 0 for x, y in zip(secret, guess)])
        b = sum((Counter(secret) & Counter(guess)).values()) - a
        return f'{a}A{b}B'
